src/openrct2-ui/UiContext.Win32.cpp

```diff
@@ -42,15 +42,25 @@
 static std::wstring SHGetPathFromIDListLongPath(LPCITEMIDLIST pidl)
 {
     std::wstring pszPath(MAX_PATH, 0);
-    while (!SHGetPathFromIDListW(pidl, &pszPath[0]))
+#if defined(__MINGW32__)
+    SHGetPathFromIDListW(pidl, pszPath.data())
```

Comment:
This looks wrong. It’s lacking a semicolon at the end. It’s also not entirely clear to me why the ifdef is there, and especially why it is different to both the native Windows code and to what was there before.

Reply:
The mingw CI doesn't have the Ex function I am trying to use, so I have to restrict that version to a MAX_PATH limit.

Reply:
I get that, but why can’t it use the old code that was here? It seems the MinGW variant now does a lot less than it did before?

Reply:
I just said:

> What was there before was just wrong, the non-Ex version of the function can not provide a longer path because it doesn't accept a buffer size.

So the loop is pointless, thus removed.